core/services/chainlink/application.go

```diff
@@ -231,7 +272,14 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 
 		registrySyncer, err := registrysyncer.New(
 			globalLogger,
-			externalPeerWrapper,
+			func() (p2ptypes.PeerID, error) {
+				p := externalPeerWrapper.GetPeer()
```

Comment:
Why are we creating two registry syncers?

Reply:
I'm guessing it has something to do with new functionality, but seems confusing. Can we add comments clearly explaining the need for these two?

Reply:
CC @cedric-cordenier

Reply:
It comes down to the fact that the registrysyncer needs a peerID, and the peerID it needs is different depending on whether we're servicing CCIP or Keystone (in the Keystone case it's the external peer, for CCIP it's not necessary).

@makramkd Is this always going to be the case, or do you see a situation where the syncer might need to know what node it is (that's what the peerID is used for)?

That being said, we should only have one registrysyncer running at any given time.